azurerm/resource_arm_dns_cname_record.go

```diff
@@ -54,112 +54,112 @@ func resourceArmDnsCNameRecord() *schema.Resource {
-		return fmt.Errorf("Error creating DNS CName Record: %s", createResponse.Error)
+	props := dns.RecordSetProperties{
+		Metadata:    metadata,
+		TTL:         &ttl,
+		CnameRecord: &cnameRecord,
 	}
 
-	readRequest := rivieraClient.NewRequest()
-	readRequest.Command = &dns.GetCNAMERecordSet{
-		Name:              d.Get("name").(string),
-		ResourceGroupName: d.Get("resource_group_name").(string),
-		ZoneName:          d.Get("zone_name").(string),
+	parameters := dns.RecordSet{
+		Name:                &name,
+		RecordSetProperties: &props,
 	}
 
-	readResponse, err := readRequest.Execute()
+	//last parameter is set to empty to allow updates to records after creation
+	// (per SDK, set it to '*' to prevent updates, all other values are ignored)
+	resp, err := dnsClient.CreateOrUpdate(resGroup, zoneName, name, dns.CNAME, parameters, eTag, "")
 	if err != nil {
-		return fmt.Errorf("Error reading DNS CName Record: %s", err)
+		return err
 	}
-	if !readResponse.IsSuccessful() {
-		return fmt.Errorf("Error reading DNS CName Record: %s", readResponse.Error)
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read DNS CNAME Record %s (resource group %s) ID", name, resGroup)
 	}
 
-	resp := readResponse.Parsed.(*dns.GetCNAMERecordSetResponse)
-	d.SetId(resp.ID)
+	d.SetId(*resp.ID)
 
 	return resourceArmDnsCNameRecordRead(d, meta)
 }
 
 func resourceArmDnsCNameRecordRead(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
+	dnsClient := meta.(*ArmClient).dnsClient
 
 	id, err := parseAzureResourceID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	readRequest := rivieraClient.NewRequestForURI(d.Id())
-	readRequest.Command = &dns.GetCNAMERecordSet{}
+	resGroup := id.ResourceGroup
+	name := id.Path["CNAME"]
+	zoneName := id.Path["dnszones"]
 
-	readResponse, err := readRequest.Execute()
+	result, err := dnsClient.Get(resGroup, zoneName, name, dns.CNAME)
 	if err != nil {
-		return fmt.Errorf("Error reading DNS A Record: %s", err)
+		return fmt.Errorf("Error reading DNS CNAME record %s: %v", name, err)
 	}
-	if !readResponse.IsSuccessful() {
-		log.Printf("[INFO] Error reading DNS A Record %q - removing from state", d.Id())
+	if result.Response.StatusCode == http.StatusNotFound {
 		d.SetId("")
-		return fmt.Errorf("Error reading DNS A Record: %s", readResponse.Error)
+		return nil
 	}
 
-	resp := readResponse.Parsed.(*dns.GetCNAMERecordSetResponse)
+	props := *result.RecordSetProperties
+	cNameRecord := *props.CnameRecord
+	cName := *cNameRecord.Cname
 
-	d.Set("name", resp.Name)
-	d.Set("resource_group_name", id.ResourceGroup)
-	d.Set("zone_name", id.Path["dnszones"])
-	d.Set("ttl", resp.TTL)
-	d.Set("record", resp.CNAMERecord.CNAME)
+	d.Set("name", name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("zone_name", zoneName)
+	d.Set("ttl", *result.TTL)
+	d.Set("etag", *result.Etag)
+	d.Set("record", cName)
 
-	flattenAndSetTags(d, &resp.Tags)
+	flattenAndSetTags(d, result.Metadata)
```

Comment:
>  do you prefer the provider match the nomenclature in the tf file or that in the sdk?

so we should be aiming for consistency with the schema (i.e. what's used in the `*.tf` file) - rather than renaming this to match the SDK; providing the field we send to/from the API is actually tags (albeit renamed), then that's fine :) i.e. users should be able to upgrade from resources made with the Riviera SDK to resources made with the Azure SDK without any code changes :)

(from what I can see, if this is tags then I'd consider this inconsistency some design feedback for the Azure API's, as the rest of the Azure API's refer to this field as `tags`)